### DISCUSSION TITLE
Nushell Env Config Error

### DISCUSSION THREAD

**Discussion by berensn** [JanDeDobbeleer/oh-my-posh#5944] [Troubleshoot]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

After installing nushell Oh-My-Posh fails to load properly, throwing a config error regarding powerline
![Screenshot from 2024-11-25 23-56-30](https://github.com/user-attachments/assets/9013c3a4-968f-43d3-88de-c2e64ab22b7d)

Here is the .oh-my-posh.nu file
![Screenshot from 2024-11-25 23-57-43](https://github.com/user-attachments/assets/c83a24d7-e0a3-4b31-99c6-c138a2dcbeea)
Line 2 is the problem, I can't edit it directly as it just changes back every time the terminal is restarted. What do I need to add to the env.nu file to set line 2 properly?
Also line 4, what values can be defined for that and what do I need to add to the env.nu file to set it properly?

Oh-My-Posh runs fine in my zsh shell, which indicates that Oh-My-Posh itself is not the issue, its nushell that is the issue. 

### Theme

`{
    "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    "blocks": [
        {
            "alignment": "left",
            "newline": true,
            "segments": [
                {
                    "style": "plain",
                    "type": "text",
                    "template": "</br> "
                }
            ],
            "type": "prompt"
        },
        {
            "alignment": "left",
            "segments": [
                {
                    "background": "#a313a8",
                    "foreground": "#b8aec8",
                    "style": "plain",
                    "template": "\u26a1 ",
                    "type": "root"
                },
                {
                    "background": "#d30cb8",
                    "foreground": "#270633",
                    "leading_diamond": "\ue0c7",
                    "properties": {
                        "folder_icon": "\uf6d7",
                        "folder_separator_icon": "<transparent> \ue0bd </>",
                        "home_icon": "\uf7db",
                        "style": "agnoster_short"
                    },
                    "style": "diamond",
                    "template": " <transparent>\ue0bd\ue0bd</> {{ .PWD }} ",
                    "trailing_diamond": "\ue0bc",
                    "type": "path"
                },
                {
                    "background": "#5da4a6",
                    "background_templates": [
                        "{{ if or (.Working.Changed) (.Staging.Changed) }}#6df1d8{{ end }}",
                        "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#5da4a6{{ end }}",
                        "{{ if gt .Ahead 0 }}#5da4a6{{ end }}",
                        "{{ if gt .Behind 0 }}#5da4a6{{ end }}"
                    ],
                    "foreground": "#000000",
                    "leading_diamond": "\ue0ba",
                    "properties": {
                        "branch_icon": "\ue725 ",
                        "fetch_stash_count": true,
                        "fetch_status": true
                    },
                    "style": "diamond",    
                    "template": "<transparent>\ue0bd\ue0bd</> {{ .UpstreamIcon }}{{ .HEAD }}{{ .BranchStatus }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }}<#ef5350> \uf046 {{ .Staging.String }}</>{{ end }} ",
                    "trailing_diamond": "\ue0bc",
                    "type": "git"
                },
                {
                    "background": "#3c345c",
                    "foreground": "#b8aec8",
                    "leading_diamond": "\ue0ba",
                    "properties": {
                        "style": "dallas",
                        "threshold": 0
                    },
                    "style": "diamond",
                    "template": "<transparent>\ue0bd\ue0bd</> {{ .FormattedMs }}s",
                    "trailing_diamond": "\ue0c6",
                    "type": "executiontime"
                },
                {
                    "background": "#910000",
                    "foreground": "#ffffff",
                    "leading_diamond": "  \ue0b2",
                    "style": "diamond",
                    "template": "<transparent>\ue0b3 \uf12a</> {{ .Meaning }} <transparent>\ue0b1</>",
                    "trailing_diamond": "\ue0b0",
                    "type": "exit"
                }
            ],
            "type": "prompt"
        },
        {
            "alignment": "right",
            "segments": [
                {
                    "background": "#29315A",
                    "foreground": "#43CCEA",
                    "leading_diamond": "\ue0c7",
                    "style": "diamond",
                    "template": " {{ .UserName }} ",
                    "type": "session"
                },
                {
                    "background": "#29315A",
                    "foreground": "#3EC669",
                    "style": "diamond",
                    "properties": {
                        "time_format": "15:04:05" //"Mon 2-Jan-2006 15:04:05-0700"
                    },
                    "template": "<transparent>\ue0bd\ue0bd</> {{ .CurrentDate | date .Format }} ",
                    "trailing_diamond": "\ue0c6",
                    "type": "time"
                }
            ],
            "type": "prompt"
        },
        {
            "alignment": "left",
            "newline": true,
            "segments": [
                {
                    "foreground": "#f718ff",
                    "foreground_templates": [
                        "{{ if gt .Code 0 }}#ff0000{{ end }}"
                    ],
                    "properties": {
                        "always_enabled": true
                    },
                    "style": "plain",
                    "template": ":\ue0bd\ue0bd \u25bd ",
                    "type": "exit"
                }
            ],
            "type": "prompt"
        }

    
    ],
    "console_title_template": "{{if .Root}} \u26a1 {{end}}{{.Folder | replace \"~\" \"🏚\" }} @ {{.HostName}}",
    "osc99": true,
    "version": 2
}`

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

nu

### Log output

```Shell
Version: 13.9.0

Segments:

ConsoleTitle(true) -   7 ms - nu in xi
session(true)      -   0 ms -  xi 
path(true)         -   9 ms -   ~ 
git(false)         -   0 ms - 
root(false)        -   0 ms - 
exit(true)         -   0 ms -   
node(false)        -   0 ms - 
go(false)          -   0 ms - 
python(false)      -   0 ms - 
shell(true)        -   0 ms - in nu 
time(true)         -   0 ms - at 00:05:05

Run duration: 46.841265ms

Cache path: /home/xi/.cache/oh-my-posh

Config path: 

Logs:

[DEBUG] 00:05:05.552 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.553 shell.go:Getenv(XDG_CACHE_HOME) - 26.029783ms
[TRACE] 00:05:05.553 shell_unix.go:CachePath() - 26.069547ms
[DEBUG] 00:05:05.553 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.553 shell.go:Getenv(POSH_THEME) - 5.811µs
[TRACE] 00:05:05.553 shell.go:resolveConfigPath() - 13.114µs
[TRACE] 00:05:05.553 shell.go:Init() - 26.980708ms
[TRACE] 00:05:05.553 shell.go:Flags() - 50ns
[TRACE] 00:05:05.554 config.go:loadConfig() - 26.821µs
[TRACE] 00:05:05.554 shell.go:Flags() - 40ns
[DEBUG] 00:05:05.554 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.554 shell.go:Getenv(OMP_CACHE_DISABLED) - 3.126µs
[TRACE] 00:05:05.554 shell_unix.go:Root() - 601ns
[DEBUG] 00:05:05.561 shell.go:Shell:587 → process name: nu
[TRACE] 00:05:05.561 shell.go:Shell() - 7.434311ms
[TRACE] 00:05:05.561 shell.go:ErrorCode() - 121ns
[DEBUG] 00:05:05.561 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.561 shell.go:Getenv(WSL_DISTRO_NAME) - 3.506µs
[TRACE] 00:05:05.561 shell_unix.go:IsWsl() - 7.004µs
[TRACE] 00:05:05.561 shell.go:GOOS() - 70ns
[DEBUG] 00:05:05.561 shell.go:Pwd:370 → NO DATA
[TRACE] 00:05:05.561 shell.go:Pwd() - 21.23µs
[TRACE] 00:05:05.561 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.561 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.561 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.561 shell.go:GOOS() - 41ns
[DEBUG] 00:05:05.561 shell.go:User:491 → xi
[TRACE] 00:05:05.561 shell.go:User() - 3.687µs
[DEBUG] 00:05:05.561 shell.go:Host:503 → pop-os
[TRACE] 00:05:05.561 shell.go:Host() - 7.874µs
[TRACE] 00:05:05.561 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.561 shell.go:TemplateCache() - 7.549778ms
[TRACE] 00:05:05.561 shell.go:Flags() - 51ns
[DEBUG] 00:05:05.561 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.561 shell.go:Pwd() - 4.038µs
[TRACE] 00:05:05.561 shell.go:GOOS() - 40ns
[DEBUG] 00:05:05.562 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.562 shell.go:Getenv(SSH_CONNECTION) - 3.136µs
[DEBUG] 00:05:05.562 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.562 shell.go:Getenv(SSH_CLIENT) - 2.795µs
[TRACE] 00:05:05.562 shell.go:TemplateCache() - 100ns
[TRACE] 00:05:05.562 shell.go:TemplateCache() - 160ns
[TRACE] 00:05:05.562 shell.go:Shell() - 141ns
[DEBUG] 00:05:05.562 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.562 shell.go:Pwd() - 20.308µs
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[DEBUG] 00:05:05.562 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.562 shell.go:Pwd() - 3.737µs
[TRACE] 00:05:05.562 shell.go:Shell() - 80ns
[TRACE] 00:05:05.562 shell.go:Shell() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:PathSeparator() - 80ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:PathSeparator() - 40ns
[TRACE] 00:05:05.562 shell.go:PathSeparator() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:PathSeparator() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:PathSeparator() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.562 shell.go:PathSeparator() - 41ns
[DEBUG] 00:05:05.562 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.562 shell.go:Pwd() - 6.172µs
[DEBUG] 00:05:05.562 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.562 shell.go:Getenv(WSL_DISTRO_NAME) - 3.367µs
[TRACE] 00:05:05.562 shell_unix.go:IsWsl() - 6.833µs
[TRACE] 00:05:05.562 shell.go:StackCount() - 80ns
[TRACE] 00:05:05.562 shell_unix.go:DirIsWritable(/home/xi) - 4.739µs
[TRACE] 00:05:05.562 shell.go:TemplateCache() - 110ns
[TRACE] 00:05:05.562 shell.go:TemplateCache() - 161ns
[TRACE] 00:05:05.563 shell.go:Shell() - 101ns
[DEBUG] 00:05:05.571 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.571 shell.go:Pwd() - 6.562µs
[TRACE] 00:05:05.571 shell.go:GOOS() - 50ns
[DEBUG] 00:05:05.572 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.572 shell.go:Getenv(WSL_DISTRO_NAME) - 16.612µs
[TRACE] 00:05:05.572 shell_unix.go:IsWsl() - 20.178µs
[TRACE] 00:05:05.572 shell_unix.go:IsWsl2() - 23.284µs
[TRACE] 00:05:05.572 shell.go:GOOS() - 40ns
[DEBUG] 00:05:05.572 shell.go:CommandPath:542 → /usr/bin/git
[TRACE] 00:05:05.572 shell.go:CommandPath(git) - 33.392µs
[TRACE] 00:05:05.572 shell.go:HasCommand(git) - 37.24µs
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.507µs
[ERROR] 00:05:05.572 shell.go:HasParentFilePath:675 → stat /.git: no such file or directory
[TRACE] 00:05:05.572 shell.go:HasParentFilePath(.git) - 31.79µs
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 6.152µs
[TRACE] 00:05:05.572 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.572 shell_unix.go:Root() - 440ns
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.296µs
[TRACE] 00:05:05.572 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.572 shell.go:ErrorCode() - 60ns
[TRACE] 00:05:05.572 shell.go:TemplateCache() - 120ns
[TRACE] 00:05:05.572 shell.go:TemplateCache() - 170ns
[TRACE] 00:05:05.572 shell.go:Shell() - 120ns
[TRACE] 00:05:05.572 shell.go:TemplateCache() - 160ns
[TRACE] 00:05:05.572 shell.go:Flags() - 70ns
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 7.194µs
[TRACE] 00:05:05.572 shell.go:GOOS() - 50ns
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.427µs
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.337µs
[TRACE] 00:05:05.572 shell.go:GOOS() - 40ns
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.246µs
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.366µs
[TRACE] 00:05:05.572 shell.go:GOOS() - 40ns
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.236µs
[DEBUG] 00:05:05.572 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.572 shell.go:Pwd() - 3.356µs
[TRACE] 00:05:05.573 shell.go:GOOS() - 40ns
[TRACE] 00:05:05.573 shell.go:Shell() - 90ns
[TRACE] 00:05:05.573 shell.go:Flags() - 40ns
[TRACE] 00:05:05.573 shell.go:TemplateCache() - 50ns
[TRACE] 00:05:05.573 shell.go:TemplateCache() - 160ns
[TRACE] 00:05:05.573 shell.go:Shell() - 110ns
[DEBUG] 00:05:05.573 shell.go:Pwd:350 → /home/xi
[TRACE] 00:05:05.573 shell.go:Pwd() - 3.997µs
[TRACE] 00:05:05.573 shell.go:GOOS() - 50ns
[TRACE] 00:05:05.573 shell.go:TemplateCache() - 90ns
[TRACE] 00:05:05.573 shell.go:TemplateCache() - 200ns
[TRACE] 00:05:05.573 shell.go:Shell() - 130ns
[DEBUG] 00:05:05.573 shell.go:Getenv:342 → NO DATA
[TRACE] 00:05:05.573 shell.go:Getenv(XDG_CACHE_HOME) - 3.797µs
[TRACE] 00:05:05.573 shell_unix.go:CachePath() - 21.671µs
[TRACE] 00:05:05.573 shell.go:Flags() - 50ns
```

**Comment by JanDeDobbeleer** [comment]
@berensn which nushell version is this? I'm using it fulltime myself but don't see this error.

**Comment by JanDeDobbeleer** [comment]
@berensn your Oh My Posh version is `13.9.0`, please upgrade first and make sure nushell is also up-to-date.

**Comment by berensn** [comment]
So in zsh, my other shell, the brew path is set in .zshrc using 'eval "$(/home/linuxbrew/.linuxbrew/bin/brew shellenv)"' allowing me to user brew in zsh. In nu, to use brew to update Oh-My-Posh I set the env.path to brews location, but it was only worked for the session. 
Once I added "$env.PATH = ($env.PATH | split row (char esep) | prepend '/home/linuxbrew/.linuxbrew/bin')" to the env.nu file everything started working fine. 

Is there a way to change the prompt indicator from the vertical bar to something like a block or blinking block?

**Reply by JanDeDobbeleer** [reply]
That's a question for the terminal. The cursor is the responsibility of the terminal.

**Comment by berensn** [comment]
Alright, thanks. 
Thanks also for looking at my issue.
